### PR TITLE
[16.0][IMP] rma: rma group

### PR DESCRIPTION
**Pull request by chafique-delli**
Some improvements regarding the RMA Group:
 - make the state visible 
 - handle the case of 'cancelled' lines for compute the state
 - addition of action buttons that we have on the RMA lines
 @CarlosVForgeFlow , @florian-dacosta , @AaronHForgeFlow

### CODE REVIEW

File: rma/models/rma_order.py [hunk 1]
```diff
@@ -54,7 +54,12 @@ def _compute_state(self):
             rma_line_to_approve = self.env["rma.order.line"].search_count(
                 [("id", "in", rec.rma_line_ids.ids), ("state", "=", "to_approve")]
             )
-            if rma_line_done != 0:
+            rma_line_no_cancel = rec.rma_line_ids.filtered(
```
Comment (sebastienbeau) [inline rma/models/rma_order.py]: We can improve the code and fix the behaviour, group should be done when everything is done

```
states = set(rec.rma_line_ids.mapped("state"))

if states == {"done"}:
    rec.state = "done"
elif states == {"cancel"}:
    rec.state = "cancel"
elif "draft" in states:
    rec.state = "draft"
elif "to_approved" in states:
    rec.state = "to_approved"
elif "draft" in states:
    rec.state = "approved"
```

File: rma/models/rma_order.py [hunk 2]
```diff
@@ -300,6 +306,31 @@ def action_view_supplier_lines(self):
                 result["res_id"] = related_lines[0]
         return result
 
+    def action_rma_to_approve(self):
+        for rec in self:
+            rec.rma_line_ids.action_rma_to_approve()
```
Comment (sebastienbeau) [inline rma/models/rma_order.py]:
```suggestion
        return self.rma_line_ids.action_rma_to_approve()
```

File: rma/models/rma_order.py (hunk 1 above)
Comment (chafique-delli) [reply rma/models/rma_order.py]: @sebastienbeau , it's done.

File: rma/models/rma_order.py (hunk 2 above)
Comment (chafique-delli) [reply rma/models/rma_order.py]: @sebastienbeau , it's done.

File: rma/models/rma_order.py [hunk 3]
```diff
@@ -45,24 +45,17 @@ def _compute_line_count(self):
...
-                [("id", "in", rec.rma_line_ids.ids), ("state", "=", "to_approve")]
-            )
-            if rma_line_done != 0:
-                state = "done"
-            elif rma_line_approved != 0:
-                state = "approved"
-            elif rma_line_to_approve != 0:
-                state = "to_approve"
+            states = set(rec.rma_line_ids.mapped("state"))
+            if states == {"cancel"}:
```
Comment (florian-dacosta) [inline rma/models/rma_order.py]: When you create the group and you did not add line yet, the state is Done and it  should not.
I guess you could first check if there are at least one rma line. If not, it should be draft.

**Review by florian-dacosta — CHANGES_REQUESTED** [review]

File: rma/models/rma_order.py (hunk 3 above)
Comment (chafique-delli) [reply rma/models/rma_order.py]: @florian-dacosta , it's fixed.

**Review by florian-dacosta — APPROVED** [review]

**Review by AaronHForgeFlow — APPROVED** [review]
LGTM. Merging

**Pull request merged**